kv/kvserver: TestRebalancingAndCrossRegionZoneSnapshotMetrics failed

kv/kvserver.TestRebalancingAndCrossRegionZoneSnapshotMetrics [failed](https://mesolite.cluster.engflow.com/invocations/default/e550608c-ec51-4bda-8566-bb27ef1a57fc?testReportRun=1&testReportShard=18&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-26.1 @ [ec883a1305620cb561809f29464430ecfd2ade2c](https://github.com/cockroachdb/cockroach/commits/ec883a1305620cb561809f29464430ecfd2ade2c):


```
        	            	expected: map[string]kvserver_test.snapshotBytesMetrics{"":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:599}, ".cross-region":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:599}, ".cross-zone":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:0}, ".rebalancing":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:599}, ".recovery":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:0}}
        	            	actual  : map[string]kvserver_test.snapshotBytesMetrics{"":kvserver_test.snapshotBytesMetrics{sentBytes:822, rcvdBytes:599}, ".cross-region":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:599}, ".cross-zone":kvserver_test.snapshotBytesMetrics{sentBytes:822, rcvdBytes:0}, ".rebalancing":kvserver_test.snapshotBytesMetrics{sentBytes:822, rcvdBytes:599}, ".recovery":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:0}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	  (string) "": (kvserver_test.snapshotBytesMetrics) {
        	            	-  sentBytes: (int64) 0,
        	            	+  sentBytes: (int64) 822,
        	            	   rcvdBytes: (int64) 599
        	            	@@ -10,3 +10,3 @@
        	            	  (string) (len=11) ".cross-zone": (kvserver_test.snapshotBytesMetrics) {
        	            	-  sentBytes: (int64) 0,
        	            	+  sentBytes: (int64) 822,
        	            	   rcvdBytes: (int64) 0
        	            	@@ -14,3 +14,3 @@
        	            	  (string) (len=12) ".rebalancing": (kvserver_test.snapshotBytesMetrics) {
        	            	-  sentBytes: (int64) 0,
        	            	+  sentBytes: (int64) 822,
        	            	   rcvdBytes: (int64) 599
        	Test:       	TestRebalancingAndCrossRegionZoneSnapshotMetrics/first_receiver
--- FAIL: TestRebalancingAndCrossRegionZoneSnapshotMetrics/first_receiver (0.00s)
=== RUN   TestRebalancingAndCrossRegionZoneSnapshotMetrics/sender
    replica_learner_test.go:2420: 
        	Error Trace:	pkg/kv/kvserver/replica_learner_test.go:2420
        	Error:      	Not equal: 
        	            	expected: map[string]kvserver_test.snapshotBytesMetrics{"":kvserver_test.snapshotBytesMetrics{sentBytes:1421, rcvdBytes:0}, ".cross-region":kvserver_test.snapshotBytesMetrics{sentBytes:599, rcvdBytes:0}, ".cross-zone":kvserver_test.snapshotBytesMetrics{sentBytes:822, rcvdBytes:0}, ".rebalancing":kvserver_test.snapshotBytesMetrics{sentBytes:1421, rcvdBytes:0}, ".recovery":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:0}}
        	            	actual  : map[string]kvserver_test.snapshotBytesMetrics{"":kvserver_test.snapshotBytesMetrics{sentBytes:599, rcvdBytes:0}, ".cross-region":kvserver_test.snapshotBytesMetrics{sentBytes:599, rcvdBytes:0}, ".cross-zone":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:0}, ".rebalancing":kvserver_test.snapshotBytesMetrics{sentBytes:599, rcvdBytes:0}, ".recovery":kvserver_test.snapshotBytesMetrics{sentBytes:0, rcvdBytes:0}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	  (string) "": (kvserver_test.snapshotBytesMetrics) {
        	            	-  sentBytes: (int64) 1421,
        	            	+  sentBytes: (int64) 599,
        	            	   rcvdBytes: (int64) 0
        	            	@@ -10,3 +10,3 @@
        	            	  (string) (len=11) ".cross-zone": (kvserver_test.snapshotBytesMetrics) {
        	            	-  sentBytes: (int64) 822,
        	            	+  sentBytes: (int64) 0,
        	            	   rcvdBytes: (int64) 0
        	            	@@ -14,3 +14,3 @@
        	            	  (string) (len=12) ".rebalancing": (kvserver_test.snapshotBytesMetrics) {
        	            	-  sentBytes: (int64) 1421,
        	            	+  sentBytes: (int64) 599,
        	            	   rcvdBytes: (int64) 0
        	Test:       	TestRebalancingAndCrossRegionZoneSnapshotMetrics/sender
--- FAIL: TestRebalancingAndCrossRegionZoneSnapshotMetrics/sender (0.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=18</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRebalancingAndCrossRegionZoneSnapshotMetrics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62206

--- a/pkg/kv/kvserver/replica_learner_test.go
+++ b/pkg/kv/kvserver/replica_learner_test.go
@@ -2340,6 +2340,12 @@ func TestRebalancingAndCrossRegionZoneSnapshotMetrics(t *testing.T) {
 
 	defer tc.Stopper().Stop(ctx)
 
+	// Disable delegated snapshots as they can screw up the metric assertions
+	// this test makes.
+	kvserver.NumDelegateLimit.Override(
+		ctx, &tc.Server(0).ClusterSettings().SV, 0,
+	)
+
 	scratchStartKey := tc.ScratchRange(t)
 	desc := tc.LookupRangeOrFatal(t, scratchStartKey)
 	// Wait for the expiration lease to upgrade to an epoch or leader lease.

